pkgs/by-name/id/idris2/unwrapped.nix

```diff
@@ -0,0 +1,144 @@
+{
+  lib,
+  stdenv,
+  chez,
+  chez-racket,
+  clang,
+  gmp,
+  installShellFiles,
+  gambit,
+  nodejs,
+  zsh,
+  callPackage,
+  idris2Packages,
+  libidris2_support,
+  idris2-version,
+  idris2-src,
+}:
+let
+  inherit (stdenv.hostPlatform) extensions;
+
+  # Runtime library
+  libsupportLib = lib.makeLibraryPath [ libidris2_support ];
+  libsupportShare = lib.makeSearchPath "share" [ libidris2_support ];
+
+  platformChez =
+    if (stdenv.system == "x86_64-linux") || (lib.versionAtLeast chez.version "10.0.0") then
+      chez
+    else
+      chez-racket;
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "idris2";
+  version = idris2-version;
+  src = idris2-src;
+
+  postPatch = ''
+    shopt -s globstar
+
+    # Patch all occurences of the support lib with an absolute path so it
+    # works without wrapping.
+    substituteInPlace **/*.idr \
+      --replace-quiet "libidris2_support" "${libidris2_support}/lib/libidris2_support${extensions.sharedLibrary}"
+
+    # The remove changes libidris2_support.a to /nix/store/..../libidris2_support.so.a
+    # Fix that up so the reference-counted C backend works
+    substituteInPlace src/Compiler/RefC/CC.idr \
+      --replace-fail "libidris2_support${extensions.sharedLibrary}.a" "libidris2_support.a"
+
+    patchShebangs --build tests
+  '';
+
+  strictDeps = true;
+  nativeBuildInputs = [
+    clang
+    platformChez
+    installShellFiles
+  ]
+  ++ lib.optionals stdenv.hostPlatform.isDarwin [ zsh ];
+  buildInputs = [
+    platformChez
+    gmp
+    libidris2_support
+  ];
+
+  enableParallelBuilding = true;
+
+  makeFlags = [
+    "PREFIX=${placeholder "out"}"
+    "IDRIS2_SUPPORT_DIR=${libsupportLib}"
+  ]
+  ++ lib.optional stdenv.hostPlatform.isDarwin "OS=";
+
+  # The name of the main executable of pkgs.chez is `scheme`
+  buildFlags = [
+    "bootstrap"
+    "SCHEME=scheme"
+    "IDRIS2_LIBS=${libsupportLib}"
+    "IDRIS2_DATA=${libsupportShare}"
+  ];
+
+  doCheck = false;
+  checkTarget = "test";
+  nativeCheckInputs = [
+    gambit
+    nodejs
+  ];
+  checkFlags = [
+    "INTERACTIVE="
+    "IDRIS2_DATA=${libsupportShare}"
+    "IDRIS2_LIBS=${libsupportLib}"
+    "TEST_IDRIS2_DATA=${libsupportShare}"
+    "TEST_IDRIS2_LIBS=${libsupportLib}"
+    "TEST_IDRIS2_SUPPORT_DIR=${libsupportLib}"
+  ];
+
+  postInstall = ''
+    # Remove existing idris2 wrapper that sets incorrect LD_LIBRARY_PATH
+    rm $out/bin/idris2
+
+    # The only thing we need from idris2_app is the actual binary
+    mv $out/bin/idris2_app/idris2${extensions.sharedLibrary} $out/bin/idris2
```

Comment:
I would never have expected someone to guess this but I also didn't think to mention it when I commented on the file extension of the support shared library: This is `idris2.so` even on Darwin systems. The shared C library built with standard C tooling is `dylib`, but Chez Scheme still outputs an `so` file extension.

Reply:
interesting

Reply:
This got marked resolved but the code has not changed here yet; just figured I would make sure there was not a change you'd made locally and not pushed up to GitHub yet.

Reply:
Oops, it is fixed in one of the commits but shuffling files around made it disappear in a later commit